### PR TITLE
Add pre-commit, style linting with ruff

### PR DESCRIPTION
**Pull request by wlach**
Again, mostly copied from the approach used in
flask-debugtoolbar.

### CODE REVIEW

File: flask_mail.py [hunk 1]
```diff
@@ -1,51 +1,36 @@
...
-if PY3:
-    string_types = str,
-    text_type = str
-    from email import policy
-    message_policy = policy.SMTP
-else:
-    string_types = basestring,
-    text_type = unicode
-    message_policy = None
-
```
Comment (wlach) [inline flask_mail.py:47]: Snipping all of this out and/or inlining it, we only support Python 3.8+ now

File: flask_mail.py [hunk 2]
```diff
@@ -55,83 +40,78 @@ def __init__(self, obj, *args):
...
-                s = text_type(bytes(s), encoding, errors)
+                s = str(s)
         else:
             s = s.decode(encoding, errors)
     except UnicodeDecodeError as e:
         if not isinstance(s, Exception):
-            raise FlaskMailUnicodeDecodeError(s, *e.args)
+            raise FlaskMailUnicodeDecodeError(s, *e.args) from e
         else:
-            s = ' '.join([force_text(arg, encoding, strings_only,
```
Comment (wlach) [inline flask_mail.py:88]: This line did not actually work afaict. Indicates that `force_text` is not properly tested.

**Comment by aenglander** [comment]
@dependabot rebase

File: flask_mail.py [hunk 3]
```diff
@@ -55,83 +40,78 @@ def __init__(self, obj, *args):
...
-                if isinstance(s, bytes):
-                    s = text_type(s, encoding, errors)
-                else:
-                    s = text_type(s)
-            elif hasattr(s, '__unicode__'):
-                s = s.__unicode__()
+        if not isinstance(s, str):
+            if isinstance(s, bytes):
+                s = str(s, encoding, errors)
             else:
```
Comment (aenglander) [inline flask_mail.py:58]: This creates some convoluted logic if left as is. Let's refactor this section to clean up the type checks.

File: flask_mail.py [hunk 4]
```diff
@@ -333,73 +325,67 @@ def _message(self):
...
+                "Content-Disposition", attachment.disposition, filename=filename
+            )
 
             for key, value in attachment.headers.items():
                 f.add_header(key, value)
 
             msg.attach(f)
-        if message_policy:
-            msg.policy = message_policy
+        msg.policy = policy.SMTP
```
Comment (aenglander) [inline flask_mail.py:380]: Why did we make this change?

**Review by aenglander — CHANGES_REQUESTED** [review]

File: flask_mail.py (hunk 4 above)
Comment (wlach) [reply flask_mail.py:380]: `message_policy` was a global that always evaluated to `policy.SMTP` in Python 3+:

https://github.com/pallets-eco/flask-mail/blob/5fdf9a03bd26342558f25c1cf8586647b167dd10/flask_mail.py#L42

File: flask_mail.py (hunk 3 above)
Comment (wlach) [reply flask_mail.py]: Hopefully this is better, at least it removes the nested ifs:

https://github.com/pallets-eco/flask-mail/pull/208/commits/1125b53887e687568f209e515745756281fafcde

I'd rather err on not doing bigger refactorings in this PR. Can always do that later.

Comment (wlach) [reply flask_mail.py]: Made the if clauses a little more intuitive: https://github.com/pallets-eco/flask-mail/pull/208/commits/e2a674ed6f7540ec9d77acb6b6dcf22546a533c0

**Pull request merged**